### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by Setsugennoao**
Update build scripts to be easier to use for developers.

* Add single `compile_all.bat`
* Add `install_deps.bat` which clones dependencies and on sequential runs pulls new commits (gets called by `compile_all.bat`)
* Add and check multiple install paths other than `C:`
* Fix py.exe / py ambiguity (yes, with multiple installs it breaks...)
* Add vsgenstubs4 to the portable version, alongside vsrepo.

Potentially breaking changes
* Rename `make portable.bat` to `make_portable.bat`
* Remove `doc/Makefile` (substitued with sphinx commands)

### CODE REVIEW

**Comment by cid-chan** [comment]
I do not own a windows system anymore. I cannot comment on this PR.

**Comment by RivenSkaye** [comment]
I tested the build process. Everything seems to work quite nicely, though the updated scripts seem to be looking for Visual Studio 2022 rather than 2019 as the documentation lists. Docs would need to be reworked for this for sure though.

If I'm not missing anything, this only needs the Python header files, so would it be possible for the automatic script to also pull in the required tarballs off the Python [dot] org archives and unpack them locally using whatever 7-zip it grabs? That would prevent the need for people to install 4 concurrent Python versions (because now I'm stuck uninstalling 3)

**Comment by Setsugennoao** [comment]
> If I'm not missing anything, this only needs the Python header files, so would it be possible for the automatic script to also pull in the required tarballs off the Python [dot] org archives and unpack them locally using whatever 7-zip it grabs? That would prevent the need for people to install 4 concurrent Python versions (because now I'm stuck uninstalling 3)

I personally don't like the idea, plus that's assuming you only want to compile the vsscript project.
In reality you 110% will also compile the cython extention and that requires the corresponding python version with all the dependencies installed to compile.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	url = https://github.com/AviSynth/AviSynthPlus.git
+[submodule "libp2p"]
+	path = libp2p
+	url = https://github.com/sekrit-twc/libp2p.git
+[submodule "vsrepo"]
+	path = vsrepo
+	url = https://github.com/vapoursynth/vsrepo.git
+[submodule "zimg"]
+	path = zimg
+	url = https://github.com/sekrit-twc/zimg.git
```
Comment (cid-chan) [inline .gitmodules:12]: Iirc, Myr used this once, and was dissatisfied.
I don't like the feature either. You're gonna have to find another solution for that.

**Review by cid-chan — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (Setsugennoao) [reply .gitmodules:12]: This isn't a solution to anything, the build scripts grab em from elsewhere, it's just that i think it's just nornal to link to other repos in github too.
If myrsloik doesn't like it for w/e reason we can just delete the file

**Pull request merged**